replicator/run.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+import re
+
+from time import sleep
+
+import dns.message, dns.query, dns.rdatatype
+import requests
+
+
+catalog_domain = 'catalog.internal.'
+master_ip = '172.16.7.3'
+config = {
+    'base_url': 'http://ns:8081/api/v1/servers/localhost',
+    'headers': {
+        'Accept': 'application/json',
+        'User-Agent': 'desecslave',
+        'X-API-Key': os.environ['DESECSLAVE_NS_APIKEY'],
+    },
+}
+
+
+class PDNSException(Exception):
+    def __init__(self, response=None):
+        self.response = response
+        return super().__init__(f'pdns response code: {response.status_code}, pdns response body: {response.text}')
+
+
+def pdns_id(name):
+    # See also pdns code, apiZoneNameToId() in ws-api.cc (with the exception of forward slash)
+    if not re.match(r'^[a-zA-Z0-9_.-]+$', name):
+        raise ValueError('Invalid hostname ' + name)
+
+    name = name.translate(str.maketrans({'/': '=2F', '_': '=5F'}))
+    return name.rstrip('.') + '.'
+
+
+def pdns_request(method, *, path, body=None):
```

Comment:
a pdns client library for python would clean up some of the code in this file as well as avoid code duplication to desec-stack/api

Reply:
Yep, for sure. That is a good idea!